templates.go

```diff
@@ -1,1606 +1,411 @@
+        --node-ip=${DEFAULT_IPV4} \
+        --api-servers=https://{{.Cluster.Kubernetes.API.Domain}} \
+        --containerized \
+        --enable-server \
+        --logtostderr=true \
+        --machine-id-file=/rootfs/etc/machine-id \
+        --cadvisor-port=4194 \
+        --healthz-bind-address=${DEFAULT_IPV4} \
+        --healthz-port=10248 \
+        --cluster-dns={{.Cluster.Kubernetes.DNS.IP}} \
+        --cluster-domain={{.Cluster.Kubernetes.Domain}} \
+        --network-plugin=cni \
+        --register-node=true \
+        --register-schedulable=false \
+        --allow-privileged=true \
+        --kubeconfig=/etc/kubernetes/config/kubelet-kubeconfig.yml \
+        --node-labels="role=master,kubernetes.io/hostname=${HOSTNAME},ip=${DEFAULT_IPV4},{{.Cluster.Kubernetes.Kubelet.Labels}}" \
+        --v=2"
+        ExecStop=-/usr/bin/docker stop -t 10 $NAME
+        ExecStopPost=-/usr/bin/docker rm -f $NAME
+        [Install]
+        WantedBy=multi-user.target
+    - name: kubeadm-init.service
+      enable: true
+      contents: |
+        [Unit]
+        Description=Initialize master with kubeadm
+        Wants=kubelet.service
+        [Service]
+        Type=oneshot
+	Environment=IMAGE=maru/kubeadm
+        Environment=NAME=%p.service
+        EnvironmentFile=/etc/network-environment
+        ExecStartPre=-/usr/bin/docker stop $NAME
+        ExecStartPre=-/usr/bin/docker rm $NAME
+        ExecStartPre=-/usr/bin/docker pull $IMAGE
+        ExecStart=/usr/bin/docker run \
+            --name $NAME \
+            $IMAGE \
+            kubeadm init --token {{.Token}}
+        [Install]
+        WantedBy=multi-user.target
+    - name: wait-for-k8s-nodes.service
+      enable: true
+      contents: |
+        [Unit]
+        Description=Wait for nodes availability
+        Wants=kubeadm-init.service
+        [Service]
+        Type=oneshot
+        Environment=IMAGE={{.Cluster.Kubernetes.Kubectl.Docker.Image}}
+	Environment=NAME=%p.service
+	EnvironmentFile=/etc/network-environment
+	ExecStartPre=-/usr/bin/docker stop $NAME
+        ExecStartPre=-/usr/bin/docker pull $NAME
+        ExecStart=/usr/bin/docker run \
+            --name $NAME \
+            $IMAGE \
+            kubectl get nodes
+        [Install]
+        WantedBy=multi-user.target
+    - name: calico-setup.service
+      enable: true
+      contents: |
+        [Unit]
+        Description=Setup self-hosted Calico
+        Wants=wait-for-k8s-nodes.service
+        [Service]
+        Type=oneshot
+        Environment=IMAGE={{.Cluster.Kubernetes.Kubectl.Docker.Image}}
+	Environment=NAME=%p.service
+	EnvironmentFile=/etc/network-environment
+	ExecStartPre=-/usr/bin/docker stop $NAME
+        ExecStartPre=-/usr/bin/docker pull $NAME
+        ExecStart=/usr/bin/docker run \
+            --name $NAME \
+            $IMAGE \
+            kubectl apply -f http://docs.projectcalico.org/v2.3/getting-started/kubernetes/installation/hosted/kubeadm/1.6/calico.yaml
```

Comment:
Is this the recommended way to do it? Just wondering if it is cool to rely on some other source which is also HTTP. I think it would be better to put the manifests in here directly.